keps/sig-storage/1495-volume-populators/README.md

```diff
@@ -80,19 +80,38 @@ problem is that current API validation logic uses a white list of object types,
 feature gates for each object type. This approach won't scale to generic populators, which
 will by their nature be too numerous and varied.
 
-This proposal recommends that we relax validation (in core Kubernetes) on the `DataSource`
-field to allow arbitrary object types to be data sources, and rely on new controllers to
-perform the validation on those data sources and provide feedback to users. We will introduce
-a new CRD to register valid datasource kinds, and individual volume populators will
-create a CR for each kind of data source that it understands.
+For historical reasons (not intentional), the validation of the `DataSource` field caused
+unsupported objects to be ignored, rather than rejected, and for the request to yield an
+empty volume, as if no data source had been requested. It's not possible to update this
+behavior in a backwards-compatible way, which forces us to use a different mechanism for
+populators.
+
+This proposal recommends that we replace the `DataSource` field with a new `DataSourceRef`
+field with expanded semantics. The existing `DataSource` field will be deprecated and
```

Comment:
> The existing `DataSource` field will be deprecated 

So "DataSource" will be deprecated in 1.22?  Does that mean it will be removed in a future release?  "DataSource" is used by GA features VolumeSnapshot and Clone.  I don't think we should remove that.

Reply:
The deprecation won't happen until the new field reaches beta or GA. In any case, it will NEVER be removed -- we would stop documenting it, and we would start generating API warnings to clients that continued to use it. The intention is that things would migrate off the old field completely at some point, but as a GA feature we wouldn't ever break something that depended on it.

Reply:
It won't be marked as deprecated in 1.22. It will be marked as deprecated once `DataSourceRef` is GA, and will never be removed. Instead there will be some admission control logic that ensures backwards compatibility (populates DataSource when DataSourceRef points to snapshot or clone).

Reply:
Sounds good.